src/parser.rs

```diff
@@ -0,0 +1,287 @@
+        }
+    }
+
+    fn parse_postfix_expression(&self) {
+        self.parse_primary_expression();
+
+        loop {
+            if self.try_peek_op(Operator.Dot) {
+                self.read_identifier();
+
+                if (self.try_peek_op(Operator.LeftParen)) {
+                    self.parse_parenthesized_argument_list();
+                    // method call: EXPR.ident(arg1, arg2, ...)
+                }
+                else {
+                    // member reference: EXPR.ident
+                }
+            }
+            else if self.try_peek_op(Operator.LeftBrace) {
+                self.read_op(Operator.LeftBrace);
+                self.parse_expression();
+                self.read_op(Operator.RightBrace);
+                // array access: EXPR[EXPR]
+            }
+            else {
+                break;
+            }
+        }
+    }
+
+    fn parse_primary_expression(&self) {
+        if self.try_peek_keyword(Keyword.Null) {
+            self.read_keyword(Keyword.Null);
+        }
+        else if self.try_peek_keyword(Keyword.False) {
+            self.read_keyword(Keyword.False);
+        }
+        else if self.try_peek_keyword(Keyword.True) {
+            self.read_keyword(Keyword.True);
+        }
+        else if self.try_peek_integer() {
+            self.read_integer();
+        }
+        else if self.try_peek_identifier() {
+            self.read_identifier();
+
+            if self.try_read_op(Operator.LeftParen) {
+                // function call
+                self.parse_parenthesized_argument_list();
+            }
+            else {
+                // var ref
+            }
+        }
+        else if self.try_peek_keyword(Keyword.This) {
+            self.read_keyword(Keyword.This);
+        }
+        else if self.try_peek_op(Operator.LeftParen) {
+            // parenthesized expression
+            self.read_op(Operator.LeftParen);
+            self.parse_expression();
+            self.read(Operator.RightParen);
+        }
+        else if self.try_peek_keyword(Keyword.New) {
+            self.read_keyword(Keyword.New);
+
+            self.parse_basic_type();
+
+            if true {
+                // todo: "new int()" is not allowed
+                // new object expression
+                self.read_op(Operator.LeftParen);
+                self.read(Operator.RightParen);
+            }
+            else {
+                // new array expression
+                self.read_op(Operator.LeftBrace);
+                self.parse_expression();
+                self.read_op(Operator.RightBrace);
+
+                while self.try_peek_op(Operator.LeftBrace) { // [
+                    self.read_op(Operator.LeftBrace);
+                    self.read_op(Operator.RightBrace);
+                }
+            }
+        }
+        else {
+            self.fail();
+        }
+    }
+
+    fn parse_parenthesized_argument_list() {
+        self.read_op(Operator.LeftParen);
+        loop {
```

Comment:
That won't work, because of commata. I think the same error has `parse_parameters()`